### PR TITLE
LCAM-1239-adding integration test for the update-crown-court endpoint

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1239)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by skasthuri36326 — APPROVED** [review]

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java [hunk 1]
```diff
@@ -202,6 +207,78 @@ void givenAValidContent_whenUpdateApplicationIsInvoked_thenUpdateApplicationIsSu
...
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    void givenAInvalidContent_whenUpdateIsInvoked_thenFailsBadRequest() throws Exception {
+        mvc.perform(RequestBuilderUtils.buildRequestGivenContent(
+                        HttpMethod.PUT, objectMapper.writeValueAsString(
+                                TestModelDataBuilder.getApiUpdateApplicationRequest(Boolean.FALSE)), ENDPOINT_URL + UPDATE_CC_URL))
+                .andExpect(status().isBadRequest());
+    }
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java:224]: Do we really need these tests? They seem to just be duplicates of what we have in the controller test class.

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java [hunk 2]
```diff
@@ -202,6 +207,78 @@ void givenAValidContent_whenUpdateApplicationIsInvoked_thenUpdateApplicationIsSu
...
+        ApiUpdateApplicationRequest apiUpdateApplicationRequest = TestModelDataBuilder.getApiUpdateApplicationRequest(Boolean.TRUE);
+        apiUpdateApplicationRequest.setCaseType(CaseType.APPEAL_CC);
+        stubForUpdateCC();
+        mvc.perform(RequestBuilderUtils.buildRequestGivenContent(
+                        HttpMethod.PUT, objectMapper.writeValueAsString(apiUpdateApplicationRequest), ENDPOINT_URL + UPDATE_CC_URL))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.crownCourtSummary.repOrderDecision").value("Granted - Passed Means Test"))
+                .andExpect(jsonPath("$.crownCourtSummary.repType").value("Crown Court Only"))
+                .andExpect(jsonPath("$.crownCourtSummary.evidenceFeeLevel").value("LEVEL1"))
+                .andExpect(jsonPath("$.crownCourtSummary.repOrderCrownCourtOutcome[0].outcome").value("SUCCESSFUL"))
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java:238]: Can these not use the existing Enums here?

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java [hunk 3]
```diff
@@ -202,6 +207,78 @@ void givenAValidContent_whenUpdateApplicationIsInvoked_thenUpdateApplicationIsSu
...
+    @Test
+    void givenAInvalidContent_whenUpdateIsInvoked_thenFailsBadRequest() throws Exception {
+        mvc.perform(RequestBuilderUtils.buildRequestGivenContent(
+                        HttpMethod.PUT, objectMapper.writeValueAsString(
+                                TestModelDataBuilder.getApiUpdateApplicationRequest(Boolean.FALSE)), ENDPOINT_URL + UPDATE_CC_URL))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void givenAValidContent_whenUpdateIsInvoked_thenUpdateIsSuccess() throws Exception {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java:227]: This constitutes the 'happy path', but what about the not so happy path? The other tests mirror what we have in the controller, so don't seem to be adding much. Personally, I'd like to see at least one more scenario covering for example an internal validation failure and/or a failing call to a downstream service.

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java (hunk 1 above)
Comment (loganathanMOJ) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java]: Thanks. Reused this test case for ValidationException

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java (hunk 2 above)
Comment (loganathanMOJ) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingIntegrationTest.java]: Thanks. Done now

**Review by Mhowell494 — APPROVED** [review]
